skimage/color/colorconv.py

```diff
@@ -824,8 +824,12 @@ def gray2rgb(image, alpha=None):
     Raises
     ------
     ValueError
-        If the input is not a 2- or 3-dimensional image.
+        If the input is not a 1-, 2- or 3-dimensional image.
 
+    Notes
+    -----
+    If the input is a 1-dimensional image of shape ``(M, )``, the output
+    will be shape ``(M, 1, 3)``.
```

Comment:
So for a 1D image you suggest returning a `(M, 3)` output? I certainly agree it's the most consistent approach. I was going to suggest that it might have potential for confusion, if you get what looks just like a single channel 2D image out of `gray2rgb`. But on second thoughts I think anyone doing something somewhat unusual such as passing an `(M, )` dimensional image would be expecting that result.

If everyone's happy I'll make the suggested change. (It seems outside the scope of this PR to tackle the higher dimensional cases however...)